src/Agoda.IoC.NetCore/StartupExtension.cs

```diff
@@ -34,6 +42,21 @@ public static IServiceCollection AutoWireAssembly<T>(this IServiceCollection ser
                     attr => new RegistrationContext(attr, x.ToType, isMockMode)
                 ))
                 .ToList();
+
+            if (registrations.Any(x => !x.Validation.IsValid))
+            {
+                var exceptions  = new ContainerRegistrationContextException("There are validations errors!!!");
+                exceptions
+                    .RegistrationContextExceptions
+                    .AddRange(registrations.Where(x => !x.Validation.IsValid)
```

Comment:
Somewhat nit-picking, but is there any way to reduce the amount of iterations over the registrations collection?